Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -599,17 +599,22 @@ extension SPNoteEditorViewController: InterlinkProcessorPresentationContextProvi
 }
 
 
-// MARK: - InterlinkProcessorDatasource
+// MARK: - InterlinkProcessorDelegate
 //
-extension SPNoteEditorViewController: InterlinkProcessorDatasource {
+extension SPNoteEditorViewController: InterlinkProcessorDelegate {
 
-    var interlinkExcudedEntityID: NSManagedObjectID? {
+    func excludedEntityIdentifierForInterlinkProcessor(_ processor: InterlinkProcessor) -> NSManagedObjectID? {
         currentNote?.objectID
     }
+
+    func interlinkProcessor(_ processor: InterlinkProcessor, insert text: String, in range: Range<String.Index>) {
+        noteEditorTextView.insertText(text: text, in: range)
+        processor.dismissInterlinkLookup()
+    }
```

Comment:
Idea: do not to pass markdown range to the Processor and back here. Instead we can figure it out right here based on the current selection in the editor.

Reply:
(For the record!!) Discussed over Slack. Let's keep the Range (for now at least!).

Thank you for bringing this up!!